tasks/main.yml

```
@@ -1,19 +1,18 @@
 ---
-- name: Install MariaDB on RedHat Systems
-  when: ansible_os_family == 'RedHat'
-  ansible.builtin.dnf:
-    name:
-      - mariadb-server
-      - mariadb
-      - python3-PyMySQL
 
-- name: Install MariaDB on Debian Systems
-  when: ansible_os_family == 'Debian'
-  ansible.builtin.apt:
-    name:
-      - mariadb-server
-      - mariadb-client
-      - python3-pymysql
+- name: Loading platform variables
+  include_vars: "{{ item }}"
+  with_first_found:
+    - "{{ ansible_distribution }}-{{ ansible_distribution_release }}.yml"
+    - "{{ ansible_distribution }}.yml"
+    - "{{ ansible_os_family }}.yml"
+  tags: always
+
+- name: Install MariaDB
+  ansible.builtin.package:
+    name: "{{ item }}"
+    state: present
+  with_items: "{{ mariadb_packages }}"
```

Shouldn't this also work (and doesn't require a loop):

```suggestion
  ansible.builtin.package:
    name: "{{ mariadb_packages }}"
    state: present
```

As an alternative, you could even use a simple inline switch here.
That way, you wouldn't have to look up somewhere else what will be installed:

```yaml
- name: Install MariaDB on RedHat Systems
  ansible.builtin.package:
    name:
      - mariadb-server
      - mariadb
      - python3-{{ 'PyMySQL' if ansible_os_family == 'RedHat' else 'pymysql' }}
```

---

Yes, this also works. Not every module accepts lists as parameters so I had used a loop here, but the package module apparently does :) for inline switches I would have to use several because not only the python package name differs and then I can just put it into a variable because we need a distribution distinction anyways